### PR TITLE
Update sqlite to version 3.43.0

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
closes https://github.com/maplibre/maplibre-native/issues/1516

### CODE REVIEW

File: vendor/sqlite/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-3.24.0
```
Comment (SiarheiFedartsou) [inline vendor/sqlite/version.txt:1]: The latest version taken from https://www.sqlite.org/download.html

**Review by louwers — APPROVED** [review]
CI passes so I'm happy with merging this.

**Comment by SiarheiFedartsou** [comment]
Some of render tests have failed with:
```
[INFO] {mbgl-render-tes}[Database]: Resource not found in cache: http://glyphs/NotoCJK/39936-40191.pbf?ignore=NotoCJK&hat_tip=jfirebaugh
Error:  {mbgl-render-tes}[Style]: Failed to load glyph range 39936-40191 for font stack NotoCJK: Not found in offline database
[INFO] {mbgl-render-tes}[Database]: Resource not found in cache: http://glyphs/NotoCJK/39680-39935.pbf?ignore=NotoCJK&hat_tip=jfirebaugh
```

🤔

**Comment by SiarheiFedartsou** [comment]
Ah it seems we have the same in `main` as well. https://github.com/maplibre/maplibre-native/actions/runs/5979856511/job/16228168980

@louwers is it expected?

**Comment by SiarheiFedartsou** [comment]
>This would be a great way to pwn the project. But the two committed files are identical to the what is distributed by SQLite.

Not sure I fully get it tbh :)

**Comment by louwers** [comment]
@SiarheiFedartsou Those 11 render tests are a problem on `main` as well....

> Not sure I fully get it tbh :)

What I mean is, I realized (after I approved) it's really easy to 'update SQLite' but throw in some malicious code since it's not really possible to check 100k lines of code. 🙂  That is why I checked the diff.

**Pull request merged**